event_reporter/reporter/application_event_reporter.go

```diff
@@ -225,6 +222,19 @@ func (s *applicationEventReporter) StreamApplicationEvents(
 	return nil
 }
 
+func (s *applicationEventReporter) resolveApplicationVersions(ctx context.Context, a *appv1.Application, logCtx *log.Entry) *apiclient.ApplicationVersions {
+	syncRevision := utils.GetOperationStateRevision(a)
+	var applicationVersions *apiclient.ApplicationVersions
+	if syncRevision != nil {
```

Comment:
lets rid of else condition and make it nicer. also you dont need applicationVersions as variable